### PR TITLE
Add examples for Artemis connector and fix issues in tests

### PR DESCRIPTION
**Pull request by riyafa**
## Purpose
-  Add examples for Artemis connector 
-  Add module ReadMe for artemis connector 
-  Fix Artemis test failing at 90% disk space 
- Delete data folder on mvn or gradle clean
    Artemis integration tests create a folder named 'data' in the integration tests repo. This PR makes sure to delete this folder when `mvn clean` or `gradle clean` is called.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/14397

## Approach
- Add examples and readme. 
- Change config of broker.xml
- Update clean plugin in mvn and clean task in gradle

## Samples
- Add samples simple multicast consumer and producer. 
- Add anycast consumer and producer that uses Connection and Session objects.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [x] Module documentation in Module.md files
  - [x] Ballerina By Examples

### CODE REVIEW

File: examples/artemis-anycast-session-consumer/artemis_anycast_session_consumer.bal [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import ballerina/artemis;
+import ballerina/io;
+
+// Explicitly creating the Connection and Session allows for reusability
```
Comment (praneesha) [inline examples/artemis-anycast-session-consumer/artemis_anycast_session_consumer.bal]:
```suggestion
// Create the connection explicitly. The session allows reusability
```

File: examples/artemis-anycast-session-producer/artemis_anycast_session_producer.out [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Make sure to have the ActiveMQ Artemis broker running
```
Comment (praneesha) [inline examples/artemis-anycast-session-producer/artemis_anycast_session_producer.out]:
```suggestion
# Make sure to have the ActiveMQ Artemis broker running
```
```suggestion
# Make sure to have the ActiveMQ Artemis broker running.
```

File: examples/artemis-anycast-session-producer/artemis_anycast_session_producer.out [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+# Navigate to the directory that contains the 'artemis_anycast_session_producer.bal' file and issue the
+# 'ballerina run' command.
```
Comment (praneesha) [inline examples/artemis-anycast-session-producer/artemis_anycast_session_producer.out]:
```suggestion
# 'ballerina run' command as follows.
```

File: examples/artemis-multicast-consumer/artemis_multicast_consumer.bal [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import ballerina/artemis;
+import ballerina/log;
+import ballerina/io;
+
+// Consumer listens to "my_queue" with the address "my_address"
```
Comment (praneesha) [inline examples/artemis-multicast-consumer/artemis_multicast_consumer.bal]:
```suggestion
// Consumer listens to the queue (i.e., "my_queue") with the address (i.e., "my_address").
```

File: examples/artemis-multicast-consumer/artemis_multicast_consumer.bal [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+}
+// Attaches the service to the listener.
+service artemisConsumer on new artemis:Listener({host:"localhost", port:61616}) {
+
+    // This resource is triggered when a message is received.
+    resource function onMessage(artemis:Message message) {
+        // Retrieves the message payload.
+        var payload = message.getPayload();
+        io:print("Payload is ");
+        // Checks for the type of the payload and prints the type
```
Comment (praneesha) [inline examples/artemis-multicast-consumer/artemis_multicast_consumer.bal]:
```suggestion
        // Checks for the type of the payload and prints the type
```
```suggestion
        // Checks for the type of the payload and prints the type.
```

File: examples/artemis-multicast-consumer/artemis_multicast_consumer.bal [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        } else if (payload is byte[]) {
+            io:println("byte[]");
+        } else if (payload is map<byte[]>) {
+            io:println("map<byte[]>");
+        } else if (payload is map<boolean>) {
+            io:println("map<boolean>");
+        } else if (payload is map<float>) {
+            io:println("map<float>");
+        }
+        // Prints the payload
```
Comment (praneesha) [inline examples/artemis-multicast-consumer/artemis_multicast_consumer.bal]:
```suggestion
        // Prints the payload.
```

File: examples/artemis-multicast-consumer/artemis_multicast_consumer.out [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# Make sure to have the ActiveMQ Artemis broker running
```
Comment (praneesha) [inline examples/artemis-multicast-consumer/artemis_multicast_consumer.out]:
```suggestion
# Make sure to have the ActiveMQ Artemis broker running.
```

File: examples/artemis-multicast-consumer/artemis_multicast_consumer.out [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+
+# Navigate to the directory that contains the 'artemis_multicast_consumer.bal' file and issue the
+# 'ballerina run' command.
```
Comment (praneesha) [inline examples/artemis-multicast-consumer/artemis_multicast_consumer.out]:
```suggestion
# 'ballerina run' command as follows.
```

File: examples/artemis-multicast-consumer/artemis_multicast_consumer.out [hunk 9]
```diff
@@ -0,0 +1,8 @@
...
+$ ballerina run artemis_multicast_consumer.bal
+# The ActiveMQ Artemis consumer runs as a Ballerina service and listens to the subscribed queue
```
Comment (praneesha) [inline examples/artemis-multicast-consumer/artemis_multicast_consumer.out]:
```suggestion
# The ActiveMQ Artemis consumer runs as a Ballerina service and listens to the subscribed queue.
```

File: examples/artemis-multicast-consumer/artemis_multicast_consumer.out [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+
+Initiating service(s) in 'artemis_multicast_consumer.bal'
```
Comment (praneesha) [inline examples/artemis-multicast-consumer/artemis_multicast_consumer.out]:
```suggestion
Initiating service(s) in the 'artemis_multicast_consumer.bal' file
```

File: examples/artemis-multicast-consumer/artmeis_multicast_consumer.description [hunk 11]
```diff
@@ -0,0 +1 @@
+// Message reception by an ActiveMQ Artemis consumer
```
Comment (praneesha) [inline examples/artemis-multicast-consumer/artmeis_multicast_consumer.description]:
```suggestion
// Message reception by an ActiveMQ Artemis consumer.
```

File: examples/artemis-multicast-producer/artemis_multicast_producer.bal [hunk 12]
```diff
@@ -0,0 +1,20 @@
+import ballerina/artemis;
+import ballerina/log;
+
+// Creates a producer to the address "my_address"
```
Comment (praneesha) [inline examples/artemis-multicast-producer/artemis_multicast_producer.bal]:
```suggestion
// Creates a producer to the "my_address" address.
```

File: examples/artemis-multicast-producer/artemis_multicast_producer.bal [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+artemis:Producer prod = new({host:"localhost", port:61616}, "my_address",
+  addressConfig = {routingType:artemis:MULTICAST});
+public function main() {
+    // Sends the string message to the artemis server
```
Comment (praneesha) [inline examples/artemis-multicast-producer/artemis_multicast_producer.bal]:
```suggestion
    // Sends the message as a String to the Artemis server.
```

File: examples/artemis-multicast-producer/artemis_multicast_producer.bal [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+    var err = prod->send("Hello World!");
+    if (err is error) {
+        log:printError("Error occurred while sending message", err = err);
+    }
+    // Closes the producer
```
Comment (praneesha) [inline examples/artemis-multicast-producer/artemis_multicast_producer.bal]:
```suggestion
    // Closes the producer.
```

File: examples/artemis-multicast-producer/artemis_multicast_producer.description [hunk 15]
```diff
@@ -0,0 +1,2 @@
+// A message is produced to a MULTICAST ActiveMQ Artemis address using the Ballerina Artemis Connector
+// that uses the Core API for connection.
```
Comment (praneesha) [inline examples/artemis-multicast-producer/artemis_multicast_producer.description]:
```suggestion
// that uses the Core API for the connection.
```

File: examples/artemis-multicast-producer/artemis_multicast_producer.out [hunk 16]
```diff
@@ -0,0 +1,5 @@
+# Make sure to have the ActiveMQ Artemis broker running
```
Comment (praneesha) [inline examples/artemis-multicast-producer/artemis_multicast_producer.out]:
```suggestion
# Make sure to have the ActiveMQ Artemis broker running.
```

File: examples/artemis-multicast-producer/artemis_multicast_producer.out [hunk 17]
```diff
@@ -0,0 +1,5 @@
...
+
+# Navigate to the directory that contains the 'artemis_multicast_producer.bal' file and issue the
+# 'ballerina run' command.
```
Comment (praneesha) [inline examples/artemis-multicast-producer/artemis_multicast_producer.out]:
```suggestion
# 'ballerina run' command as follows.
```

File: stdlib/messaging/activemq-artemis/src/main/ballerina/artemis/Module.md [hunk 18]
```diff
@@ -0,0 +1,48 @@
+## Module overview
+
+This module provides the capability to connect with a ActiveMQ Artemis broker using the [Core API](https://activemq.apache.org/artemis/docs/latest/core.html)
```
Comment (praneesha) [inline stdlib/messaging/activemq-artemis/src/main/ballerina/artemis/Module.md]:
```suggestion
This module provides the capability to connect with a ActiveMQ Artemis broker using the [Core API](https://activemq.apache.org/artemis/docs/latest/core.html).
```

File: stdlib/messaging/activemq-artemis/src/main/ballerina/artemis/Module.md [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+
+## Samples
+
+### Artemis Producer
+
+Following program will produce a message to the Artemis broker
```
Comment (praneesha) [inline stdlib/messaging/activemq-artemis/src/main/ballerina/artemis/Module.md]:
```suggestion
The following program produces a message to the Artemis broker.
```

File: stdlib/messaging/activemq-artemis/src/main/ballerina/artemis/Module.md [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+        err = prod->close();
+        if (err is error) {
+            log:printError("Error occurred closing producer", err = err);
+        }
+    }
+}
+```
+### Artemis consumer
+
+Following program will consume a message from an Artemis broker
```
Comment (praneesha) [inline stdlib/messaging/activemq-artemis/src/main/ballerina/artemis/Module.md]:
```suggestion
The following program will consume a message from the Artemis broker.
```

File: examples/artemis-anycast-session-consumer/artemis_anycast_session_consumer.out [hunk 21]
```diff
@@ -0,0 +1,8 @@
+# Make sure to have the ActiveMQ Artemis broker running
+
+# Navigate to the directory that contains the 'artemis_multicast_consumer.bal' file and issue the
+# 'ballerina run' command.
+$ ballerina run artemis_multicast_consumer.bal
+# The ActiveMQ Artemis consumer runs as a Ballerina service and listens to the subscribed queue
+
+Initiating service(s) in 'artemis_multicast_consumer.bal'
```
[the remark this answers is not in the document]
Comment (riyafa) [reply examples/artemis-anycast-session-consumer/artemis_anycast_session_consumer.out]: This is what is printed in the terminal from the language side code, hence cannot be changed

File: examples/artemis-anycast-session-producer/artemis_anycast_session_producer.bal [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+        "name": "John",
+        "age": "25"
+    };
+    // Creates a message object using a session using map content
+    artemis:Message msg = new(session, mapContent);
+    // Creates a producer to the address "hello" using the session object
+    artemis:Producer prod = new(session, "queue1", 
+        addressConfig = {autoCreated:false});
+
+    // Sends the string message to the artemis server
```
[the remark this answers is not in the document]
Comment (riyafa) [reply examples/artemis-anycast-session-producer/artemis_anycast_session_producer.bal]: `string` in Ballerina has a simple 's'

File: examples/artemis-multicast-consumer/artemis_multicast_consumer.out (hunk 10 above)
Comment (riyafa) [reply examples/artemis-multicast-consumer/artemis_multicast_consumer.out]: This is what is printed in the terminal from the language side code, hence cannot be changed

File: examples/artemis-multicast-producer/artemis_multicast_producer.bal (hunk 13 above)
Comment (riyafa) [reply examples/artemis-multicast-producer/artemis_multicast_producer.bal]: It is really a string type message so saying it as string message makes more sense.

**Pull request merged**